### DISCUSSION TITLE
Waveform is rendered, but cannot play on iOS

### DISCUSSION THREAD

**Discussion by mulhoon** [katspaugh/wavesurfer.js#3302] [Q&A] [answered]
Hi, 
I am using wavesurfer in a capacitor project and having trouble on iOS

I'm using the record plugin and saving the blob using [capacitor-blob-writer](https://github.com/diachedelic/capacitor-blob-writer)

I'm then loading it back and using `Filesystem.getUri` and `Capacitor.convertFileSrc(uri)` to get a file ref like this `capacitor://localhost/_capacitor_file_/var/mobile/Containers/Data/Application/6269...`

I load it in wavesurfer and it renders the waveform fine, but when I use `wavesurfer.play()` I get a browser issue 
`Unhandled Promise Rejection: NotSupportedError: The operation is not supported.` I researched this [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play#notsupportederror)

If I load the file ref as src in an `<audio>` tag it plays fine with the default html player. Just not with wavesurfer. Does anyone know why this might be?

Thanks

**Comment by katspaugh** [comment, marked as answer]
The original audio URL gets transformed into a blob URL in wavesurfer, and that's what a resulting audio element is getting in the src.
Looks like this doesn't work in your case. Try using an external audio element as described here: https://github.com/katspaugh/wavesurfer.js/discussions/3291#discussioncomment-7359876

**Comment by mulhoon** [comment]
Thank you for your swift reply. It worked! 🙏
```js
media: new Audio(file)
```
Are there any drawbacks to this? it seems to work perfectly.

**Reply by katspaugh** [reply]
With this approach it will download the audio twice (to render and to play) unless you provide peaks.

**Reply by pranay-engineears** [reply]
@mulhoon did you add the custom audio element to the DOM? Also, is the `file` url here `blob:` URL? If so, how did you get it to play successfully in just an `audio` tag in Safari?

**Reply by mulhoon** [reply]
@pranay-engineears I don't add a dom element, simply add this in the options for wavesurfer... And yes, file is the `blob:`
```js
{
  media: new Audio(file)
}
```
Is it not working for you? Are you talking Safari on desktop or iOS?

**Reply by pranay-engineears** [reply]
@mulhoon Both on safari desktop and iOS. When i put my WAV audio file blob URL into a plain audio tag like this:

```
<audio src="blob:http://localhost:3000/af4069b6-fe62-41b8-b422-2a4b441898e8" controls />
```
The media player has an error:
<img width="223" alt="Screenshot 2024-01-12 at 9 02 31 AM" src="https://github.com/katspaugh/wavesurfer.js/assets/155573445/cfda6b87-1ef6-4ef0-ba14-b3fb6601c125">

Doing the same works fine on chrome. Is there anything special you did to get your blob audio file to play successfully in a normal audio tag on safari/iOS?

**Reply by katspaugh** [reply]
This might not work if the file is served with the wrong MIME. Make sure the content type is `audio/***`

**Reply by pranay-engineears** [reply]
Got it to work by explicitly setting the MIME type of the Blob content. Thank you!
